conformance/tests/httproute-disallowed-kind.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteDisallowedKind)
+}
+
+var HTTPRouteDisallowedKind = suite.ConformanceTest{
+	ShortName:   "HTTPRouteDisallowedKind",
+	Description: "A single HTTPRoute in the gateway-conformance-infra namespace should fail to attach to a Listener that does not allow the HTTPRoute kind",
+	Manifests:   []string{"tests/httproute-disallowed-kind.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		// This test creates an additional Gateway in the gateway-conformance-infra
+		// namespace so we have to wait for it to be ready.
+		kubernetes.NamespacesMustBeReady(t, suite.Client, []string{"gateway-conformance-infra"}, 300)
+
+		routeName := types.NamespacedName{Name: "httproute-disallowed-kind", Namespace: "gateway-conformance-infra"}
+		gwName := types.NamespacedName{Name: "tlsroutes-only", Namespace: "gateway-conformance-infra"}
+
+		// TODO: Determine if this is actually what we want. It is likely
+		// preferable to have status set with some kind of warning/error message
+		// but that is also unlikely to be universally achievable.
+		t.Run("Route should not have Parents set in status", func(t *testing.T) {
+			parents := []v1alpha2.RouteParentStatus{}
+			kubernetes.HTTPRouteMustHaveParents(t, suite.Client, routeName, parents, true, 60)
+		})
```

Comment:
I don't think we need to block on this since other tests are already doing the same thing, but created https://github.com/kubernetes-sigs/gateway-api/issues/1135 to track this.